src/code/file_menus.asm

```diff
@@ -146,60 +146,66 @@ func_001_4852::
     pop  de                                       ; $4891: $D1
     ld   a, $05                                   ; $4892: $3E $05
 
-jr_001_4894::
+.drawEmptyRowLoop
+    ; Draw the empty row above the save slot name;
+    ; might contain diacritics
     ldh  [hMultiPurpose0], a                      ; $4894: $E0 $D7
     ld   a, [de]                                  ; $4896: $1A
     and  a                                        ; $4897: $A7
 
 IF LANG_EN
-    jr   .jr_489D                                 ; $4898: $18 $03
+    jr   .selectEmptyRowTile                      ; $4898: $18 $03
+    ; Unreachable code:
     dec  a                                        ; $489A: $3D
     and  $C0                                      ; $489B: $E6 $C0
 ELSE
-    jr   z, .jr_489D
+    jr   z, .selectEmptyRowTile
     dec  a
     push hl
     push bc
     ld   c, a
     ld   b, $00
-    call ReadDialogBankFromTable
-IF __PATCH_9__
+    call ReadTileValueFromDiacriticsTable
+IF __DO_CHECK_DIACRITICS__
     ldh  [hRoomBank], a
 ENDC
     pop  bc
     pop  hl
     cp   $00
 ENDC
 
-.jr_489D::
-IF __PATCH_B__ == 1
-    ld   a, $7E
-    jr   z, .jr_48A9
-    ldh  a, [hRoomBank]
-    cp   $02
-    ld   a, $C8
-    jr   z, .jr_48A9
-    inc  a
-ELIF __PATCH_B__ == 2
-    ld   a, $7E
-    jr   z, .jr_48A9
-    ld   a, $c9
+.selectEmptyRowTile::
+    ; Select what tile to draw in the empty space
+IF __DO_CHECK_DIACRITICS__
+    ld   a, $7E               ; Empty tile
+    jr   z, .drawEmptyRowTile ; Jump if no diacritic
+    ldh  a, [hRoomBank]       ; Load value from DiacriticsTable
+    cp   2                    ; Check if DiacriticsTable had value 2
+    ld   a, $C8               ; First diacritic tile
+    jr   z, .drawEmptyRowTile ; Jump if diacritic 2
+    inc  a                    ; Second diacritic tile
+ELIF LANG_FR
```

Comment:
Not a fan of re-introducing language-specific code to the codebase. The whole point of these `PATCH_` flags was to abstract away these differences.

Reply:
Yes, abstracting the languages away from the code base seems good – although for me it seems more relevant when a patch is shared by more than one language. If it's a single language, and on a small piece of code, maybe the cost of defining a patch for this is not worth it?